lua/mini/statusline.lua

```diff
@@ -281,33 +281,46 @@ MiniStatusline.section_git = function(args)
   return string.format('%s %s %s', icon, head, signs)
 end
 
+--- Section for Neovim's builtin lsp
+---
+--- Shows nothing if there is no attached lsp
+--- Uses LspAttach and LspDetach events to determine if an LSP is attached.
+---
+---@param args __statusline_args Use `args.icon` to supply your own icon.
+---
+---@return __statusline_section
+MiniStatusline.section_lsp = function(args)
```

Comment:
We mentioned this section didn't feel useful in the issue. I find its nice to know if the LSP server is working and connected, but I also get this from `mini.notify`.

Happy to either remove this, or keep it and write up unit tests.